storybook/src/Storybook/FlickableImage.qml

```diff
@@ -0,0 +1,32 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+Item {
+    property alias source: image.source
+
```

Comment:
It is good practice to set implicit sizes on reusable items.
```suggestion
        implicitWidth: image.implicitWidth
        implicitHeight: image.implicitHeight
```

Reply:
I have to disagree. Flickable doesn't have an implicit size because it doesn't know its content. I would expect that if I instantiate `FlickableImage` without the size it will automatically use the size of the loaded image same as for normal `Image`. 

One simple scenario I can think of is to have a grid of images, where we want to avoid scaling and yet have some size constraints:

```
GridLayout {
    FlickableImage { Layout.maximumWidth: 100; Layout.maximumHeight: 100; source: a }
    FlickableImage { Layout.maximumWidth: 100; Layout.maximumHeight: 100; source: b }
    FlickableImage { Layout.maximumWidth: 100; Layout.maximumHeight: 100; source: c }
    FlickableImage { Layout.maximumWidth: 100; Layout.maximumHeight: 100; source: d }
}
```

Reply:
Hmm, I have to disagree to :smile: 
> Flickable doesn't have an implicit size because it doesn't know its content.

Flickable always knows size of its content via `contentWidth`/`contentHeight`. So potentially `Flickable` could have implicit size equal to content size. It would play nicely with the example you provided. But this usage is rare case, probably for that reason qt guys decided not doing that.
> One simple scenario I can think of is to have a grid of images, where we want to avoid scaling and yet have some size constraints

That's example is partially valid. For dynamically loaded images first we would end up with `BusyIndicator`s stacked one on top another. After loading layout would be as expected. So we would have to provide one implicit size for loading and second when loaded, depending on `Image` status. For that very simple component it's quite complicated logic. Implementing it here just in case is imo overkill and against YAGNI principle.

For me implicit size makes sense for Button-like components, some panels composed of several controls, etc. Everything that has some kind of obvious size (not applicable only in some corner cases). Than it's good practice to provide implicit size. On the other hand components based on Flickable (Flickable itself, ListView, GridView) has no "natural" size in general. In some cases we could expect some default size, but those cases are rather rare, therefore it makes no so much sense to cover them up front. So in that case I decided to follow qt's convention and skipped providing implicit size for those potential use cases you indicated, especially that it's possible to make it working on client side by providing preferred size equal to content size + maximum size as you did.

I do not insist on this, because I know that it is mainly matter of interpretation of vague terms like "natural size". It's just explanation way of thinking about that :)

Reply:
> Flickable doesn't have an implicit size because it doesn't know its content.

I meant it doesn't know its content not in terms of its combined size but in terms of its purpose. Here, you know for sure `FlickableImage` is only one image, and I as a client would expect the "natural" size, especially when the root component is an `Item`.

> So we would have to provide one implicit size for loading and second when loaded, depending on Image status. For that very simple component it's quite complicated logic. Implementing it here just in case is imo overkill and against YAGNI principle.

This one I didn't take into consideration. I agree with YAGNI.

> For me implicit size makes sense for Button-like components, some panels composed of several controls, etc. Everything that has some kind of obvious size (not applicable only in some corner cases). Than it's good practice to provide implicit size.

Exactly. For example `Image` has some kind of obvious size and Qt provides that.

> On the other hand components based on Flickable (Flickable itself, ListView, GridView) has no "natural" size in general. In some cases we could expect some default size, but those cases are rather rare, therefore it makes no so much sense to cover them up front. 

That's clear.

I think the main debate here is how we understand `FlickableImage` image. For you it is a flickable with an image, for me, it is an unscalable image with scrolling capabilities.

Anyway, it has become bikeshedding now :sweat_smile: Since the logic for implicit size would be more complicated than simple one-element binding, let's align to YAGNI principle as you suggested and leave it as it.

Reply:
I agree. In general we are on the same page :) In case of using that in other context I would probably extend it as you described. Thanks for discussion!